### ISSUE TITLE
Import Helper does not respect `eslint`/`prettier` rules

### ISSUE THREAD

**Issue by tukusejssirs** [unlight/sublime-import-helper#97]
I prefer no spaces after `{` and before `}`, and no semicolon after commands. I have set `eslint` and `prettier` accordingly.

However, I could not find a way to make Import Helper respect these settings. Is there a way to configure this?

Thanks for this great plugin! :wink:

**Comment by unlight** [comment]
Sorry, no. There is only one setting [from_quote](https://github.com/unlight/sublime-import-helper#from_quote)
It is not possible to take into account all potential code fixers, stylers and formatters.

I think, you can configure your editor to apply `eslint --fix` on save.

**Comment by tukusejssirs** [comment]
> I think, you can configure your editor to apply `eslint --fix` on save.

Yes, I use that workaround, but I hoped we could insert the import correctly formatted.

---

I looked at the code, but I don’t code in Python. I’d like to change the hard-coded formatting to my preffered one (in my fork), but I was lost. Could you point me to the rignt file/line? Thanks. :wink:

**Comment by unlight** [comment]
Actually there is no formatting in python code.
Python sends code to nodejs process and library [import-adjutor](https://github.com/unlight/import-adjutor) actually do inserting.

You need to check `insertImport` command https://github.com/unlight/import-adjutor/blob/master/src/insert-import.ts

On the other hand `import-adjutor` is a wrapper around [ts-morph](https://github.com/dsherret/ts-morph)

I see that setting related to space before brace is exists there
https://github.com/dsherret/ts-morph/blob/latest/packages/ts-morph/src/options/ManipulationSettingsContainer.ts#L54

But no setting related to `no semicolon`

So, you need:
1. patch import-adjutor
2. update dependency in import-helper to use patched version
3. build import-helper

**Comment by tukusejssirs** [comment]
@unlight, I see that in 4319e8d you added `no_semicolon`—thanks! That is half of the problem.

If similarly to `no_semicolon`, we could have a setting `no_spaces_in_brackets` it would be awesome. It would solve everything I need! :smiley:

**Comment by unlight** [comment]
@tukusejssirs 
Check v4.1.0 https://github.com/unlight/sublime-import-helper/releases

**Comment by tukusejssirs** [comment]
Thanks for being so fast, @unlight!

However, I could not make it work. Package Control automatically upgraded the package to `v4.1.0`. I have the following in the package-specific settings:

```json
{
  "autocomplete_auto_import": true,
  "insert_space_in_braces": false,
  "no_semicolon": true
}
```

When I add an import using Import Helper, it still added spaces in brackets and semicolons at the end of the line. I tested this in a third-party project though, where `.eslint` is configured to to use spaces in brackets and semicolons, however you said that Import Helper does not care about the settings of `eslint`.

**Comment by unlight** [comment]
Where do you keep these settings? Package/User/import_helper.settings or in sublime-project file?

**Comment by tukusejssirs** [comment]
I open the settings file via `Preferences` → `Package Settings` → `Import Helper`, which opens the package settings on the left and user settings (overrides) on the right. The user settings file is located in `~/.config/sublime-text/Packages/User/import_helper.sublime-settings`.

**Comment by unlight** [comment]
Sorry, but I cant reproduce this.
Check settings in sublime project file, it has more priority.

**Comment by tukusejssirs** [comment]
I’ve just checked it and now it works. Maybe it required a ST restart, which I didn’t do yesterday.

Anyway, it seems like it does not check the format of previously imported stuff, e.g. `import { one } from './app';` is changed into `import { one, two} from './app'` after importing `two`.

**Comment by tukusejssirs** [comment]
@unlight, I think we can close this issue. Although the original request is not fulfilled, my needs are satisfied. Thanks! :pray:
